### DISCUSSION TITLE
无法编译附录中的参考文献

### DISCUSSION THREAD

**Discussion by wangdm17** [tuna/thuthesis#610] [Q&A] [answered]
我使用VScode中的LaTeX Workshop按照xe-bib-xe-xe的顺序编译，附录中的\cite{}始终处于？？的状态，黄标提示称“Citation `abrahams99tex' on page 17 undefined.”，附录中的图片、表格与公式等可以正常引用，正文部分也一切正常，参考文献可以正常显示，请问这是什么问题？

**Comment by zepinglee** [comment, marked as answer]
建议先阅读模板的文档。本科生附录部分的参考文献需要用 bibtex 额外编译。

```bash
$ bibtex thuthesis-example.aux              # 正文的参考文献
$ bibtex thuthesis-example-survey.aux       # 本科生的调研报告的参考文献
$ bibtex thuthesis-example-translation.aux  # 本科生的外文资料翻译的参考文献
$ bibtex thuthesis-example-index.aux        # 本科生的书面翻译对应的原文索引
```

推荐用最新版 `.vscode/settings.json` 里定义的 `latexmk for thuthesis` recipe，可以正常编译。
<img width="374" alt="Screen Shot 2021-03-13 at 19 15 18" src="https://user-images.githubusercontent.com/12290822/111028269-8c2d2e80-8430-11eb-93f2-2d6d8d7e9e90.png">

**Reply by wangdm17** [reply]
已解决，感谢
